packages/components/src/Dialog/Layout/DialogLayout.tsx

```diff
@@ -94,23 +72,19 @@ export const DialogLayout: FC<DialogLayoutProps> = ({
   const dialogFooter = (footer || footerSecondary) && (
     <DialogFooter secondary={footerSecondary}>{footer}</DialogFooter>
   )
+  const dialogHeader =
+    header &&
+    (!headerDetail ? (
+      <DialogHeader hideClose={!headerCloseButton}>{header}</DialogHeader>
+    ) : (
+      <DialogHeader detail={headerDetail}>{header}</DialogHeader>
+    ))
```

Comment:
We're trying to avoid nested-ternarys like this across the code base (there's a ticket to go unwind the existing ones eventually).

With that in mind the best approach is generally to lift this kind of code into a separate function outside of the component:

```suggestion
  const dialogHeader = constructDialogHeader(header, headerDetail, headerCloseButton)
```

```
const constructDialogHeader = (
  children?: ReactNode,
  detail?: ReactNode,
  closeButton?: boolean
) => {
  if (!children) return null
  const props: DialogHeaderProps = { children }

  if (!detail) {
    props.detail = detail
  } else {
    props.hideClose = !closeButton
  }

  return <DialogHeader {...props} />
}
```